### PR TITLE
Rompath 

### PR DESCRIPTION
**Pull request by ColumPaget**
Hiya,

This pull adds a rompath system, where the user can specify a list of directories to search in for roms using the '-rp' command-line option, or by setting the GZX_ROMPATH environment variable. 

regards,

Colum

### CODE REVIEW

**Comment by jxsvoboda** [comment]
Hi, thanks for your PR. I have concerns about this, however. A colon-sparated list won't work properly on Windows because you won't be able to specify drive as part of the path. Why do you need a list of paths, anyway? Overall, can you help me understand what is the motivation for this change? It's not useful for starting with a different working directory, because then the font wouldn't be loaded. Are you trying to use alternate roms, perhaps developing rom-based software?

**Comment by ColumPaget** [comment]
Sorry for the late reply, I only found the notification today. I'd not considered the issue of windows, but in general the reason for having a rompath is simply being able to start up a program (in this case spectrum game) from anywhere. Otherwise one has to go to the directory with the game roms in it, and that also needs to contain the base spectrum roms. So if gzx is being shipped in a distro, then you need to have a wrapper script or something to chdir to the appropriate directory, and then run the program there. Also a rompath allows you to break your roms up into different directories, and not have, potentially, thousands of speccy roms sharing a directory. 

So basically, with a rompath one can invoke a spectrum game from anywhere, from a desktop menu or whatever, without having to do the 'first chdir to a romdir' dance.

Colum

**Comment by jxsvoboda** [comment]
I don't think rompath is a good solution for that problem. GZX needs different resource files for its operation. Not just the ROM files, but also a font file (font.bin), palette file (sp256.pal) and possibly other files in the future and you need to find all of these. Also, if you introduce such an option (e.g. where to find resource files), you just replace the problem of having to cd to a particular directory with the problem of having to pass option(s) each time. So it's no good for linux style file system anyway.

For the case where we are located under a particular directory you could try to find the files in the same dir as from where the binary was executed from. Some DOS/Windows software does this. Other just wants you to start wit the right working directory and perhaps creates some BAT files to do so as part of the installation process.

IMHO if you are starting the emulator from a desktop UI, be it Windows or GNOME or something, you can set up the shortcut/Icon to set the working directory for you. If you are running it from the command line, you can create a script or alias to do this for you.

Maybe this is more of a lack of installation process than a gap in the emulator itself.

Also if I really saw a need to load an alternate ROM (which I don't see right now), I'd consider more important and useful to be able to do it at run-time via an UI than having an option to do that at startup). Unlike say FUSE I think all functionality should be available via UI, rather than command-line options (which can be only applied by restarting the program).

Having said that, I don't think this PR can be accepted as it stands.
